pyoptsparse/pyOpt_optimizer.py

```diff
@@ -60,34 +62,33 @@ def __init__(
             caseSensitiveOptions=caseSensitiveOptions,
         )
         self.callCounter = 0
-        self.sens = None
-        self.optProb = None
-        self.version = version
+        self.sens: Union[None, Callable, Gradient] = None
+        self.optProb: Optimization
```

Comment:
I think this is the only other place where the initialization is missing.

Reply:
This is a bit tricky because the rest of the code relies on `self.optProb` being set, so `mypy` will complain that `NoneType` has no attribte `setDVs` etc. In the future, I would like to refactor the code which will solve these problems, but I think this is okay for now.